Modules/Sources/WordPressShared/Utility/Instrumentation.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import CocoaLumberjackSwift
+
+// MARK: - Assertions
+
+/// The system that logs assertions in production.
+public protocol AssertionLogging {
+    func trackAssertion(message: String, filename: String, line: UInt, userInfo: [String: Any]?)
+}
+
+public enum AssertionLoggerDependencyContainer {
+    /// - warning: Not thread-safe. Set it once and early in the program.
+    public static var logger: AssertionLogging?
+}
+
+/// Logs an assertion. If compiled using debug configuration, crashes.
+public func wpAssert(_ closure: @autoclosure () -> Bool, _ message: StaticString = "–", userInfo: [String: Any]? = nil, file: StaticString = #file, line: UInt = #line) {
+    guard !closure() else {
+        return
+    }
+
+    let filename = (file.description as NSString).lastPathComponent
+    DDLogError("WPAssertionFailure in \(filename)–\(line): \(message)\n\(userInfo ?? [:])")
+
+    if let logger = AssertionLoggerDependencyContainer.logger,
+        WPAssertion.shouldSendAssertion(withID: "\(filename)-\(line)") {
```

Comment:
I made a slight change in behavior:  `WPAssertion.shouldSendAssertion` will apply for both Tracks and Sentry, which is fine.